src/web/app/components/question-responses/grouped-responses/grouped-responses.component.spec.ts

```diff
@@ -31,10 +107,111 @@ describe('GroupedResponsesComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(GroupedResponsesComponent);
     component = fixture.componentInstance;
+    component.session = mockSession;
+    component.responses = [mockQuestionOutput];
+    component.userToEmail = {
+      'giver-1': 'giver1@example.com',
+      'recipient-1': 'recipient1@example.com',
+    };
+    component.instructorCommentTableModel = {
+      'response-1': {
+        isAddingNewComment: false,
+        isReadOnly: false,
+        commentRows: [],
+        newCommentRow: {
+          commentEditFormModel: {
+            commentText: '',
+            isUsingCustomVisibilities: false,
+            showCommentTo: [],
+            showGiverNameTo: [],
+          },
+          isEditing: false,
+        },
+      },
+    };
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('hasRealResponses: should return true when there are real responses', () => {
+    component.ngOnInit();
+    expect(component.hasRealResponses).toBeTruthy();
+  });
+
+  it('hasRealResponses: should return false when all responses are missing', () => {
+    const missingResponse = { ...mockResponse, isMissingResponse: true };
+    const missingQuestionOutput = {
+      ...mockQuestionOutput,
+      allResponses: [missingResponse],
+    };
+    component.responses = [missingQuestionOutput];
+    component.ngOnInit();
+    expect(component.hasRealResponses).toBeFalsy();
+  });
+
+  it('teamInfo: should return correct team info for GRQ mode', () => {
+    component.isGrq = true;
+    const teamInfo = component.teamInfo;
+    expect(teamInfo['recipient']).toBe('(Team 2)');
+    expect(teamInfo['giver']).toBe('(Team 1)');
+  });
+
+  it('teamInfo: should return correct team info for RGQ mode', () => {
+    component.isGrq = false;
+    const teamInfo = component.teamInfo;
+    expect(teamInfo['recipient']).toBe('(Team 2)');
+    expect(teamInfo['giver']).toBe('(Team 1)');
+  });
+
+  it('teamInfo: should handle empty team name', () => {
+    const emptyTeamResponse = { ...mockResponse, recipientTeam: '' };
+    const emptyTeamQuestionOutput = {
+      ...mockQuestionOutput,
+      allResponses: [emptyTeamResponse],
+    };
+    component.responses = [emptyTeamQuestionOutput];
+    component.isGrq = true;
```

Comment:
Similar to above, setting `isGrq` does not impact the results. Can we remove this?